src/iop/expfusion.c

```diff
@@ -0,0 +1,1268 @@
+{
+  if(darktable.gui->reset) return;
+  dt_iop_fusion_params_t *p = (dt_iop_fusion_params_t *)self->params;
+
+  p->exposure_width = dt_bauhaus_slider_get(widget);
+
+  dt_dev_add_history_item(darktable.develop, self, TRUE);
+}
+
+static void _weight_mode_callback(GtkWidget *widget, dt_iop_module_t *self)
+{
+  if(darktable.gui->reset) return;
+  dt_iop_fusion_params_t *p = (dt_iop_fusion_params_t *)self->params;
+
+  p->weight_mode = dt_bauhaus_combobox_get(widget);
+
+  dt_dev_add_history_item(darktable.develop, self, TRUE);
+}
+
+static void _grey_projector_callback(GtkWidget *widget, dt_iop_module_t *self)
+{
+  if(darktable.gui->reset) return;
+  dt_iop_fusion_params_t *p = (dt_iop_fusion_params_t *)self->params;
+
+  p->grey_projector = dt_bauhaus_combobox_get(widget);
+
+  dt_dev_add_history_item(darktable.develop, self, TRUE);
+}
+
+static void _fusion_colorspace_callback(GtkWidget *widget, dt_iop_module_t *self)
+{
+  if(darktable.gui->reset) return;
+  dt_iop_fusion_params_t *p = (dt_iop_fusion_params_t *)self->params;
+  dt_iop_fusion_gui_data_t *g = (dt_iop_fusion_gui_data_t *)self->gui_data;
+
+  p->fusion_colorspace = dt_bauhaus_combobox_get(widget);
+
+  _show_hide_controls(p, g);
+
+  dt_dev_add_history_item(darktable.develop, self, TRUE);
+}
+
+static void _fusion_grey_projector_callback(GtkWidget *widget, dt_iop_module_t *self)
+{
+  if(darktable.gui->reset) return;
+  dt_iop_fusion_params_t *p = (dt_iop_fusion_params_t *)self->params;
+
+  p->fusion_grey_projector = dt_bauhaus_combobox_get(widget) + 1;
+
+  dt_dev_add_history_item(darktable.develop, self, TRUE);
+}
+
+static void _exposure_left_cutoff_callback(GtkWidget *widget, dt_iop_module_t *self)
+{
+  if(darktable.gui->reset) return;
+  dt_iop_fusion_params_t *p = (dt_iop_fusion_params_t *)self->params;
+
+  p->exposure_left_cutoff = dt_bauhaus_slider_get(widget) / 100.f;
+
+  dt_dev_add_history_item(darktable.develop, self, TRUE);
+}
+
+static void _exposure_right_cutoff_callback(GtkWidget *widget, dt_iop_module_t *self)
+{
+  if(darktable.gui->reset) return;
+  dt_iop_fusion_params_t *p = (dt_iop_fusion_params_t *)self->params;
+
+  p->exposure_right_cutoff = dt_bauhaus_slider_get(widget) / 100.f;
+
+  dt_dev_add_history_item(darktable.develop, self, TRUE);
+}
+
+void gui_init(struct dt_iop_module_t *self)
+{
+  self->gui_data = malloc(sizeof(dt_iop_fusion_gui_data_t));
+  dt_iop_fusion_gui_data_t *g = (dt_iop_fusion_gui_data_t *)self->gui_data;
+  dt_iop_fusion_params_t *p = (dt_iop_fusion_params_t *)self->params;
+
+  self->widget = gtk_box_new(GTK_ORIENTATION_VERTICAL, DT_BAUHAUS_SPACE);
+  dt_gui_add_help_link(self->widget, dt_get_help_url(self->op));
+
+  g->sl_num_exposures = dt_bauhaus_slider_new_with_range(self, 2, 5, 1, p->num_exposures, 0);
+  gtk_widget_set_tooltip_text(g->sl_num_exposures, _("number of different exposures to fuse"));
+  dt_bauhaus_slider_set_format(g->sl_num_exposures, "%.2fEV");
```

Comment:
The number of different exposures is an integer - it shouldn't be EV and should not be %.2f since it can only be integers